static/src/js/components/Table/Table.jsx

```diff
@@ -0,0 +1,113 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import BootstrapTable from 'react-bootstrap/Table'
+import Placeholder from 'react-bootstrap/Placeholder'
+import PlaceholderButton from 'react-bootstrap/PlaceholderButton'
+
+import For from '../For/For'
+import stringifyCircularJSON from '../../utils/stringifyCircularJSON'
+
+/**
+ * Table
+ * @typedef {Object} Table
+ * @property {Boolean} loading A value provided by whichever useQuery is being used in parent component
+ * @property {Object[]} headers A list of header titles for a table
+ * @property {Array} renderRows An array of formatted rows with data already populated
+ * @property {String} error A string with a specific error for table
+ */
+
+const Table = ({
+  headers, classNames, loading, renderRows, error
+}) => {
+  const renderHeaders = headers.map((header) => (
+    <th key={header}>{header}</th>
+  ))
+
+  const content = []
+
+  if (loading) {
+    content.push(
+      <For key="for-each-loading-key" each={[...new Array(5)]}>
+        {
+          (item, i) => (
+            <tr key={`placeholder-row_${i}`}>
+              <td className="col-md-4">
+                <Placeholder animation="glow">
+                  <Placeholder xs={6} />
+                </Placeholder>
+              </td>
+              <td className="col-md-4">
+                <Placeholder animation="glow">
+                  <Placeholder xs={8} />
+                </Placeholder>
+              </td>
+              <td className="col-md-2">
+                <Placeholder animation="glow">
+                  <Placeholder xs={8} />
+                </Placeholder>
+              </td>
+              <td className="col-auto">
+                <Placeholder animation="glow">
+                  <PlaceholderButton className="btn-sm" xs={10} variant="secondary" size="sm" />
+                </Placeholder>
+              </td>
+            </tr>
+          )
+        }
+      </For>
+    )
+  } else if (renderRows.length > 0) {
+    const rowData = renderRows.map((row) => {
+      const rowKey = stringifyCircularJSON(row)
+
+      return (
+        <tr key={`${rowKey}`}>
+          {
+            row.map((cell, index) => {
+              const cellKey = stringifyCircularJSON(cell)
+
+              return (
+                <td key={cellKey} className={classNames[index] || 'col-auto'}>
```

Comment:
yeah you are already providing a key here, so I don't think the key is needed in the caller.